core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -33,12 +33,87 @@ import fs2.internal._
 import Resource.ExitCase
 import Pull._
 
-/** A `p: Pull[F,O,R]` reads values from one or more streams, returns a
-  * result of type `R`, and produces a `Stream[F,O]` when calling `p.stream`.
+/** A pure abstraction representing the intention to perform a sequence of actions
+  * in an effect type `F`, which emits any number of values of an output type `O`.
+  * This sequence may end up in several ways:
```

Comment:
This feels a bit too detailed for the introduction -- feels like it's internal docs on the interpreter. Suggest something like: 

```
There are various ways in which a pull may terminate abnormally:
 - exceptions that occur during evaluation of a pull result
 - requested interruption of execution
```